### ISSUE TITLE
asciidoctorj should be an executable jar, which accepts same CLI options as asciidoctor 

### ISSUE THREAD

**Issue by zaxebo1** [asciidoctor/asciidoctorj#360]
(this is not urgent issue/feature, can be taken up later)

TL;DR: version=
make "asciidoctorj.jar" as executable jar (that is, which can be launched using "java -jar asciidoctorj.jar ........", which will accept all the command line parameters accepted by asciidoctor (native)

****

Details=
just like asciidoctor(native ruby) supports command line options, for example

````
   asciidoctor a1.adoc
   asciidoctor -b html5 a1.adoc
   asciidoctor --version

````

In the same way, it may be good if there is Main class in this .jar manifest , which invokes a program to parse the CLI input arguments in asciidoctorj too

````
   java -jar asciidoctorj.jar  a1.adoc
   java -jar asciidoctorj.jar -b html5 a1.adoc
   java -jar asciidoctorj.jar --version

````

All the command line options of "asciidoctor" should be also be valid for "java -jar asciidoctorj.jar"
It will be good for shell scriptors.

**Comment by zaxebo1** [comment]
HENCE, in my shell script I can just do

````
#!/bin/bash
#native asciidoctor is commented
#myasciidoctor=asciidoctor
#java based asciidoctor command is now active
myasciidoctor="java -jar asciidoctor.jar"

${myasciidoctor}   -b html5 a1.adoc

```

So, in my shell script i will be able to replace "asciidoctor" , with the "java -jar asciidoctor.jar"   - and still the command line will work as earlier

**Comment by zaxebo1** [comment]
Just now found that if i install asciidoctorj via gvmtool, then i get 

1)  shell script wrapper over asciidoctorj as $HOME/.gvm/asciidoctorj/1.5.2/bin/asciidoctorj  , which I have pasted at http://pastebin.com/kZzeD825

2)  windows batch file wrapper over asciidoctorj as $HOME/.gvm/asciidoctorj/1.5.2/bin/asciidoctorj.bat  , which I have pasted at http://pastebin.com/5p7ZPR3U

****

BUT  i was earlier  folllowing the official manual  http://asciidoctor.org/docs/asciidoctorj/#distribution  , and so when i download the .jar , then I do not got these wrapper shell scripts, to run asciidoctorj from cli.

****

I request that a "Main" class be set in the executable jar and a shell script wrapper over it should be bundled in official distribution.

**Comment by zaxebo1** [comment]
the earlier suggested above syntax

````
   java -jar asciidoctorj.jar -b html5 a1.adoc
````

can also support the better syntax

>   **java -cp asciidoctorj.jar org.asciidoctor.Main   -b html5 a1.adoc**

**Comment by mojavelinux** [comment]
As you've discovered, we do indeed make AsciidoctorJ a runnable jar, with the help of some scripts. That's what the `distZip` task in Gradle produces and is available via GVM or Bintray. You can find the link to the distribution on Bintray here: https://bintray.com/asciidoctor/maven/asciidoctorj#files (the file that ends in -bin.zip).

Unfortunately, it's not enough just to add the Main-Class to the asciidoctorj jar manifest because it also depends on several libraries (JRuby, JCommander and a logging library). We had thought about doing this at one point and opted instead to make the distribution.

With that said, you can download the distribution and run it from the lib directory without the helper scripts as follows:

```
java -cp jruby-complete-1.7.16.1.jar:asciidoctorj-1.5.2.jar:jcommander-1.35.jar:slf4j-api-1.7.7.jar:slf4j-simple-1.7.7.jar org.asciidoctor.cli.AsciidoctorInvoker file.adoc
```

I suppose we could generate a manifest that references these jar files and also specifies the main class so that within the directory you can run `java -jar asciidoctor-1.5.2.jar`.

**Comment by zaxebo1** [comment]
Wow. Now its working. Thanks.

Actually there were some extensions which were only available in asciidoctorj, not on asciidoctor. So i wanted to try this without delving into java/groovy/gradle scripts. 

The above "org.asciidoctor.cli.AsciidoctorInvoker"  is life saver for me. 
Thanks again , I can now try those extensions :-)

**Comment by mojavelinux** [comment]
:+1:

**Comment by robertpanzer** [comment]
@zaxebo1 So you're ok with closing this issue?

**Comment by zaxebo1** [comment]
yes, you can close it

**Comment by robertpanzer** [comment]
Thank you!

**Comment by anb0s** [comment]
I'm new with asciidocj and want to use it as CLI tool with "java -jar asciidoc.jar". So it does not work.
Also with the "org.asciidoctor.cli.AsciidoctorInvoker" it shows "java.lang.ClassNotFoundException: org.asciidoctor.cli.AsciidoctorInvoker".

Any hints how to use version 2.1.0 with CLI ?

**Comment by robertpanzer** [comment]
Thanks for finding this!
Can you please create a new ticket?
We moved this class to `org.asciidoctor.jruby.cli.AsciidoctorInvoker` and the `Main-Class` attribute apparently still points to the old name.

> Any hints how to use version 2.1.0 with CLI ?

If you download the zip from http://jcenter.bintray.com/org/asciidoctor/asciidoctorj/2.1.0/:asciidoctorj-2.1.0-bin.zip you will find a start script inside the archive at `bin/asciidoctorj` or `bin/asciidoctorj.bat`.

**Comment by anb0s** [comment]
> Thanks for finding this!
> Can you please create a new ticket?
ok, opened #831

**Comment by anb0s** [comment]
> If you download the zip from http://jcenter.bintray.com/org/asciidoctor/asciidoctorj/2.1.0/:asciidoctorj-2.1.0-bin.zip you will find a start script inside the archive at `bin/asciidoctorj` or `bin/asciidoctorj.bat`.

Thanks now it shows CLI options :)
The right link is: http://jcenter.bintray.com/org/asciidoctor/asciidoctorj/2.1.0/asciidoctorj-2.1.0-bin.zip
